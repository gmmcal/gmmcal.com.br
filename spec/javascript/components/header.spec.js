import { mount, shallowMount } from '@vue/test-utils'
import Header from '@/components/header'
import Menu from '@/components/menu'
import create from '../settings/utils'

describe('Header', () => {
  let props, wrapper

  beforeAll(() => {
    window.gon = {
      flag_links: create('menu_items'),
      menu_links: create('menu_items'),
    }

    props = {
      propsData: {
        image: 'bar.jpg'
      }
    }

    wrapper = mount(Header, props)
  })

  test('Component is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('Component data properties', () => {
    const expected = ['flag_links', 'menu_links', 'bannerWidth', 'bannerHeight', 'windowWidth', 'windowHeight', 'i18n_css_class', 'main_css_class', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  describe('Window', () => {
    test('is landscape', () => {
      wrapper.setData({
        windowWidth: 100,
        windowHeight: 90
      })
      expect(wrapper.vm.banner_image_class).toEqual('banner-img bgwidth')
    })

    test('is portrait', () => {
      wrapper.setData({
        windowWidth: 90,
        windowHeight: 100
      })
      expect(wrapper.vm.banner_image_class).toEqual('banner-img bgheight')
    })
  })

  describe('Child components', () => {
    test('Menu', () => {
      const child_component = wrapper.find(Menu)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    test('Renders 2 Menu objects', () => {
      wrapper = shallowMount(Header)
      const menus = wrapper.findAll(Menu)
      expect(menus.length).toBe(2)
    })
  })
})
