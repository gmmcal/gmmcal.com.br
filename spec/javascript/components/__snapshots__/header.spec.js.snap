// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div class="menu fixed-top">
    <div class="navbar navbar-expand-md navbar-dark"><button aria-label="Menu" type="button" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
      <div class="menu-container collapse navbar-collapse">
        <ul class="navbar-nav">
          <li class="active"><a data-turbolinks="false" href="#home" alt="home" class="home">Home</a></li>
          <li><a data-turbolinks="false" href="#about" alt="about" class="about">About</a></li>
        </ul>
      </div>
    </div>
    <ul class="i18n">
      <li class="active"><a data-turbolinks="false" href="#home" alt="home" class="home">Home</a></li>
      <li><a data-turbolinks="false" href="#about" alt="about" class="about">About</a></li>
    </ul>
  </div>
  <div class="banner">
    <div class="lazy banner-img">
      <div class="banner-text container">
        <h1>Gustavo Cunha</h1>
        <h2>Web developer, Mobile and Software Engineer Consultant</h2>
      </div>
    </div>
  </div>
</header>
`;
